### PR TITLE
fix/helm-chart

### LINKED ISSUE #121: Refactor: remove edc.cors config in puris-backend
During discussions we (thanks to @eschrewe) noticed that the configuration `edc.web.rest.cors.enabled` in the backend is not needed. Please remove the property from all properties files.

### LINKED ISSUE #172: Bug: Backend API Key in helm chart is hard coded
Problem:
Currently the API key is hard coded in the frontend-deployment.yaml.

Solution:
Let it be pulled based on the backend api key value.

### LINKED ISSUE #237: Enhance helm charts to use secrets
**As** puris operator,
**I want** to use the helm / kubernetes secrets engine to provide sensitive data ,
**so that** data security has been enhanced.

## Hints / Details

- Identify secrets (such as passwords, api keys and secrets) that are passed directly to the deployments and services BUT are sensitive
- Follow the kubernetes secrets documentation ([basics](https://kubernetes.io/docs/concepts/configuration/secret/), [more specific](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data)) to use the secrets mechanism
- Also refer to [portal CD secrets](https://github.com/eclipse-tractusx/portal-cd/blob/main/charts/portal/templates/secret-backend-postgres-init.yaml) usage as they show how to do it (e.g. take from secret if present, else from value)

### Outcome
- ...
### Acceptance Criteria
- Secrets can be provided to env variables via secret engine

## Out of Scope

- ...

### LINKED ISSUE #260: Insecure postgres password exposure in values.yaml
Creating issue related to hardcoded password in the values.yaml file:

At line https://github.com/eclipse-tractusx/puris/blob/146a67830519d3e5fef301ddd1e2a0c794e1c94a/charts/puris/values.yaml#L468 password is set to 'password'. Suggest to use existing postgresql secret with auto-generated password.

updates https://github.com/eclipse-tractusx/puris/issues/257

### LINKED ISSUE #268: [TRG 4.03] users' ids don't seem to match
Users' IDs set in the Dockerfile don't match ones in helm chart:

Frontend:

https://github.com/eclipse-tractusx/puris/blob/f1106401823c56b664aa796e6d6d67045650efc0/frontend/Dockerfile#L61

https://github.com/eclipse-tractusx/puris/blob/f1106401823c56b664aa796e6d6d67045650efc0/charts/puris/values.yaml#L68

Backend:

https://github.com/eclipse-tractusx/puris/blob/f1106401823c56b664aa796e6d6d67045650efc0/backend/Dockerfile#L34

https://github.com/eclipse-tractusx/puris/blob/f1106401823c56b664aa796e6d6d67045650efc0/charts/puris/values.yaml#L252

More details: https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-03
Updates https://github.com/eclipse-tractusx/puris/issues/257

### LINKED ISSUE #270: Bug: License header is before she-bash in start-nginx.sh
## Current behaviour

During [auto-merge commit](https://github.com/eclipse-tractusx/puris/commit/5bc492f7bc7498beff3a4003f095c8fe341fb3d6), the license header has been duplicated and thus, she-bang is not the first statement resulting in a crashing container.

## Expected behavior

Order must be:
- she-bang
- license
- script

## Steps to reproduce the Bug

- build frontend
- start local deployment
- check logs of frontend > error in script execution

### PR DESCRIPTION
**Pull request by PaMarzec**
## To do's
- remove hard coded backend api-key
- remove web.cors.enabled in project
- try to implement logging and env in values.yaml like [here](https://github.com/eclipse-tractusx/tractusx-edc/blob/main/charts/tractusx-connector-memory/values.yaml)
- add https in backend-deployment.yaml endpoints, if ingress and tls are enabled
- add secrets for passwords 

## Description
fixes #172
fixes #270 
resolves #121 
resolves #237
resolves #260
resolves #268  

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: charts/puris/templates/backend-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "backend.fullname" . }}
+  namespace: {{ .Release.Namespace | default "default" | quote }}
```
[the remark this answers is not in the document]
Comment (PaMarzec) [reply charts/puris/templates/backend-configmap.yaml]: Removed "namespace" key in backend-configmap and frontend-configmap in force push after this notice

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Please adapt logging or just rely on env for logging

**Comment by tom-rm-meyer-ISST** [comment]
@almadigabor or @tomaszbarwicki: As stated in pr #145, we didn't run the helm test itself. Now it fails and says the Chart.yaml is missing. I cross-checked with the [trace-x example](https://github.com/eclipse-tractusx/traceability-foss/blob/9d49d92ffc6a7121d6e0a9cc35dee22fc943331b/.github/workflows/helm-test.yaml), which finds the Chart.yaml.

Any ideas?

File: charts/puris/values.yaml [hunk 2]
```diff
@@ -353,35 +357,33 @@ backend:
         timeoutSeconds: 1
 
     puris:
+        # -- Secret for backend passwords. For more information look into 'backend-secrets.yaml' file.
+        secret: "secret-backend-puris"
```
Comment (tom-rm-meyer-ISST) [inline charts/puris/values.yaml:361]: Why not use `existingSecret`, same as for postgres

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
When installing the chart, the postgres doesn't start. 

When checking the pod (`kubectl describe pod -n puris chart-test-backend-postgresql-0`), I get the following error:
`arning  Failed     11s (x4 over 38s)  kubelet            Error: couldn't find key password in Secret puris/secret-postgres-init`

File: charts/puris/values.yaml [hunk 3]
```diff
@@ -449,13 +455,17 @@
...
         database: "postgres"
-        # -- Username for the database.
+        # -- Secret containing the password. For more information look into 'backend-secrets-postgres.yaml' file.
+        existingSecret: "secret-postgres-init"
+        # -- Username for the root user and for the database.
         username: "puris"
-        # -- Password for the database.
-        password: "password"
+        # -- Password for the root user. Secret-key 'password'
+        password: ""
```
[the remark this answers is not in the document]
Comment (PaMarzec) [reply charts/puris/values.yaml:469]: Due to the coding of the backend-secrets-postgres.yaml file a password will be generated automatically:
image

Example in deployment:
image

**Review by tomaszbarwicki — COMMENTED** [review]
hi @tom-rm-meyer-ISST actually tested it locally and it doesn't work too, getting the same issue as the workflow. 
The problem as I believe is related to backend being unable to access the secret hence failing connection to postgres.
I can't find in the backend deployment related configuration for the secret. Therefore I suggest adding one way of distribution following: https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure depending how you want to consume that in the app

File: charts/puris/templates/backend-deployment.yaml [hunk 4]
```diff
@@ -46,23 +46,35 @@ spec:
             - name: DATASOURCE_USERNAME
               value: "{{ .Values.postgresql.auth.username }}"
             - name: DATASOURCE_PASSWORD
-              value: "{{ .Values.postgresql.auth.password }}"
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.postgresql.auth.existingSecret }}"
+                  key: "postgres-password"
```
Comment (tomaszbarwicki) [inline charts/puris/templates/backend-deployment.yaml:52]: hi @tom-rm-meyer-ISST I tested it locally and get the same result as workflow run, backend fails connection to postgres.
I think this is the reason, key should refer to "password".

File: charts/puris/templates/backend-deployment.yaml [hunk 5]
```diff
@@ -46,23 +46,35 @@ spec:
...
+            - name: ROOT_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.postgresql.auth.existingSecret }}"
+                  key: "password"
```
Comment (tomaszbarwicki) [inline charts/puris/templates/backend-deployment.yaml:57]: and here key: "postgres-password"

**Review by tomaszbarwicki — CHANGES_REQUESTED** [review]

File: charts/puris/values.yaml [hunk 6]
```diff
@@ -446,7 +452,11 @@ postgresql:
   auth:
     # -- Name of the database.
     database: "postgres"
-    # -- Username for the database.
+    # -- Secret containing the password. For more information look into 'backend-secrets-postgres.yaml' file.
+    existingSecret: "secret-postgres-init"
+    # -- Username for the root user and for the database.
```
Comment (tomaszbarwicki) [inline charts/puris/values.yaml:457]: Not sure why you call it root user, it's a custom user creation as per docs: https://github.com/bitnami/charts/blob/main/bitnami/postgresql/README.md

image

same two lines below for password.

**Review by tomaszbarwicki — APPROVED** [review]
LGTM

File: charts/puris/values.yaml (hunk 6 above)
Comment (tom-rm-meyer-ISST) [reply charts/puris/values.yaml]: updated, thanks for pointing out

**Review by tom-rm-meyer-ISST — APPROVED** [review]

**Pull request merged**